iotedgedev/azurecli.py

```diff
@@ -157,24 +157,35 @@ def get_default_subscription(self):
 
     def get_subscription_id_starts_with(self, token):
         with output_io_cls() as io:
-            result = self.invoke_az_cli_outproc(["account", "list", "--query", "[?starts_with(@.id,'{0}') || starts_with(@.name,'{0}')] | [0]".format(token)],
+            result = self.invoke_az_cli_outproc(["account", "list", "--query", "[?starts_with(@.id,'{0}') || starts_with(@.name,'{1}')] | [0]".format(token.lower(), token)],
                                                 "Could not find a subscription that starts with '{0}'".format(token), io)
+
             if result:
                 out_string = io.getvalue()
                 if out_string:
+
                     data = json.loads(out_string)
-                    return data["id"]
+                    if len(data) == 1:
+                        return data[0]["id"]
+                    elif len(data) > 1:
+                        self.output.error("Found multiple subscriptions that start with '{0}'".format(token))
+                    else:
+                        self.output.error("Could not find a subscription that starts with '{0}'".format(token))
+
         return ''
 
     def set_subscription(self, subscription):
 
         if len(subscription) < 36:
```

Comment:
I considered adding a new issue to handle more use cases / user experience. 

For examples, 

- what if they were searching by name and happened to provide >= 36 char
- should we support contains instead of starts with
- can we make this even more user friendly, e.g., provide integer ids for each subscription and prompt to select 1, 2, 3...
- 

I decided any of those could be just as well done as separate issues after decided what we want.

Reply:
1. yes, checking bounds
2. no, want starts with because less likely chance for multiples returned
3. yes, but no easy way to do it.

Agree that these should be sep issues.

Reply:
Just a thought for 2. When searching by name, if I have "SE Account - david" and "SE Account - jon", then contains "dav" is less likely to return multiples than starts with "SE Account - ".